AE-508 update forta version

rossAE opened the pull request:
Renamed files to conform with forta-agent 0.0.18 and wrote tests for anomalous-value-js and transfer-ownership.

rbasri committed:
updated versions and wrote tests where missing

rossAE committed:
Create test

rossAE committed:
Delete test

rossAE committed:
updating forta version

rossAE committed:
Merge branch 'AE-508_update_forta_version' of github.com:arbitraryexecution/forta-examples into AE-508_update_forta_version

arbitrary-kdawg commented on README.md:
```diff
@@ -1,2 +1,3 @@
 # forta-examples
 Malicious Address Detection
+JS examples have been updated to forta agent v0.0.18
```
This looks like a changelog entry.  I would revert this change and instead, to make the README file useful, we should list all the directory names with a short description of each agent.

rossAE replied:
Done

arbitrary-kdawg commented on anomalous-value-js/src/agent.js:
```diff
@@ -6,8 +6,9 @@ const { Finding, FindingSeverity, FindingType, getJsonRpcUrl } = require("forta-
-function provideHandleTransaction(provider) {
+function provideHandleTransaction(contractAddresses, provider) {
   return async function handleTransaction(txEvent) {
+    //console.log(contractAddresses[txEvent.to]);
```
Did you intend to keep this debug code?

rossAE replied:
Nope, thanks for catching

arbitrary-kdawg commented on anomalous-value-js/src/agent.spec.js:
```diff
@@ -0,0 +1,128 @@
+  return oneGwei.times(numGwei);
+}
+
+describe("anomolus value agent", () => {
```
typo: "anomolus"

rossAE replied:
fixed

arbitrary-kdawg commented on anomalous-value-js/src/agent.spec.js:
```diff
@@ -0,0 +1,128 @@
+    return new TransactionEvent(type, network, transaction, receipt, traces, addresses, block);
+  };
+
+  //eventually put some mock rolling statistics on this test address
```
should this comment be marked TODO?

rossAE replied:
Changed to "initialize a test address ready for mock RollingMath statistics"

arbitrary-kdawg commented on anomalous-value-js/src/agent.spec.js:
```diff
@@ -0,0 +1,128 @@
+      
+      expect(findings).toStrictEqual([
+        Finding.fromObject({
+          name: "High Value",
```
"High Value" is relative--this agent detects outliers.  The name "Anomalous High Value Transaction" would be more descriptive.

rossAE replied:
Agreed, fixed

arbitrary-kdawg commented on anomalous-value-js/src/agent.spec.js:
```diff
@@ -0,0 +1,128 @@
+      expect(findings).toStrictEqual([
+        Finding.fromObject({
+          name: "High Value",
+          description: `Value: ${value}`,
```
Also include the contract address in the description
"Contract 0xABCD was sent XYZ Ether"

rossAE replied:
Done

arbitrary-kdawg commented on anomalous-value-js/src/agent.js:
```diff
@@ -6,8 +6,9 @@ const { Finding, FindingSeverity, FindingType, getJsonRpcUrl } = require("forta-
 
-function provideHandleTransaction(provider) {
+function provideHandleTransaction(contractAddresses, provider) {
   return async function handleTransaction(txEvent) {
```
I think a description of the handler would be useful here--you have to read into the code to figure out that this agent records transaction values for all contracts that it processes and identifies unusually high transaction values.

rossAE replied:
Included the following comment when `contractAddresses` is initialized: "Need to initialize a dictionary whose keys will be contract addresses the agent encounters and whose values are RollingMath objects tracking the ether values of transactions that touch the contract"

arbitrary-kdawg commented on anomalous-gas-js/src/agent.js:
```diff
@@ -1,4 +1,3 @@
-const ethers = require("ethers");
```
The code in these examples was never fully linted.  Eslint currently shows 269 issues.  After running with the --fix option, here is what's left:

```
/home/ubuntu/forta-uma-suite/src/examples/anomalous-gas-js/src/agent.js
  3:29  error  Unable to resolve path to module 'rolling-math'                          import/no-unresolved
  8:35  error  'rollingMath' is already declared in the upper scope on line 6 column 7  no-shadow

/home/ubuntu/forta-uma-suite/src/examples/anomalous-value-js/src/agent.js
   9:39  error  A constructor name should not start with a lowercase letter                    new-cap
  11:35  error  'contractAddresses' is already declared in the upper scope on line 8 column 7  no-shadow
  11:54  error  'provider' is already declared in the upper scope on line 9 column 7           no-shadow
  43:51  error  Expected '!==' and instead saw '!='                                            eqeqeq
  45:7   error  Assignment to property of function parameter 'contractAddresses'               no-param-reassign

/home/ubuntu/forta-uma-suite/src/examples/anomalous-value-js/src/agent.spec.js
   3:7   error  'RollingMath' is assigned a value but never used             no-unused-vars
  14:39  error  A constructor name should not start with a lowercase letter  new-cap

/home/ubuntu/forta-uma-suite/src/examples/big-tx-js/src/agent.js
  2:7  error  'BigNumber' is assigned a value but never used  no-unused-vars

/home/ubuntu/forta-uma-suite/src/examples/big-tx-js/src/agent.spec.js
  11:1  error  This line has a length of 109. Maximum allowed is 100  max-len

/home/ubuntu/forta-uma-suite/src/examples/deployment-blacklist-js/src/agent.js
  1:7   error  'ethers' is assigned a value but never used         no-unused-vars
  2:7   error  'BigNumber' is assigned a value but never used      no-unused-vars
  4:42  error  'getJsonRpcUrl' is assigned a value but never used  no-unused-vars

/home/ubuntu/forta-uma-suite/src/examples/deployment-blacklist-js/src/agent.spec.js
  14:1  error  This line has a length of 167. Maximum allowed is 100  max-len

/home/ubuntu/forta-uma-suite/src/examples/function-call-js/src/agent.spec.js
  68:9   error  'to' is assigned a value but never used    no-unused-vars
  68:25  error  'data' is assigned a value but never used  no-unused-vars

/home/ubuntu/forta-uma-suite/src/examples/transfer-ownership/src/agent.js
  15:27  error  Expected '!==' and instead saw '!='  eqeqeq
```

rossAE replied:
Linted!

arbitrary-kdawg commented on anomalous-value-js/src/agent.spec.js:
```diff
@@ -0,0 +1,128 @@
+      
+      expect(findings).toStrictEqual([
+        Finding.fromObject({
+          name: "High Value",
```
"High Value" is relative--this agent detects outliers.  A better name might be "Abnormally High Transaction Value"

rossAE replied:
Agreed, fixed

arbitrary-kdawg commented on anomalous-value-js/src/agent.spec.js:
```diff
@@ -0,0 +1,128 @@
+      expect(findings).toStrictEqual([
+        Finding.fromObject({
+          name: "High Value",
+          description: `Value: ${value}`,
```
Might want to include the contract address here as well:
"Transaction with contract 0xABCD had unusually high value of XYZ Ether"

rossAE replied:
Done

arbitrary-kdawg commented on transfer-ownership/src/agent.spec.js:
```diff
@@ -0,0 +1,58 @@
+  
+        expect(findings).toStrictEqual([
+            Finding.fromObject({
+                name: event.split("(")[0],
```
Add a comment in the source code that shows an example of what will be printed for 'name' and 'description'

rossAE replied:
Done

arbitrary-kdawg commented on transfer-ownership/src/agent.spec.js:
```diff
@@ -0,0 +1,58 @@
+const { handleTransaction } = require('./agent');
+
+//Take same list of monitored events as used in the agent script
+const ABI = require("./events.json");
```
Suggest changing ABI to abi and EVENTS to events just for consistency...we've been using all caps only for hardcoded basic types - integers and strings.

rossAE replied:
Done

rossAE committed:
corrected typos and quick fixes from kdawg comments

rossAE committed:
updated README.md

rossAE committed:
Changed anomalous value alert description

rossAE committed:
added handler description for anomalous value

rossAE committed:
Update README.md

rossAE committed:
linted code

rossAE committed:
Merge branch 'AE-508_update_forta_version' of github.com:arbitraryexecution/forta-examples into AE-508_update_forta_version
kdwag included license upstream

arbitrary-kdawg commented on function-call-js/src/agent.js:
```diff
@@ -19,7 +19,7 @@ async function handleTransaction(txEvent) {
     findings.push(
       Finding.fromObject({
-        name: 'USDT Transfer',
+        name: 'AE-FUNCTION-CALLED',
```
name is assigned to the alert ID

arbitrary-kdawg commented on transfer-ownership-js/package.json:
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "ownership-transfered-agent",
```
typo

arbitrary-kdawg committed:
Updated forta-agent versions to 0.0.23.
Renamed 'forta-uniswap-py' and 'forta-uniswap-event-py' directories.
Deleted transfer-ownership (old copy of transfer-ownership-js).
Updated README files.
Minor cleanup (typos, linting).
